css-shared-element-transitions/index.bs

```diff
@@ -438,34 +519,52 @@ The {{SameDocumentTransition/finished}} [=getter steps=] are to return [=this's=
         As such, the prose style is written to match other steps in that algorithm.
 </div>
 
+<div algorithm="suppress rendering">
+    Issue: Define where this sits within the [=update the rendering=] steps.
+
+    1. For each {{Document}} in <var ignore>docs</var> with a [=document/transition suppressing rendering=] of true:
+
+        Issue: Define this behavior. Lifecycle updates can still be triggered
+        via script APIs which query style or layout information but no visual updates
+        are presented to the user. Is this the same behavior as render-blocking?
+
+        Issue: How should input be handled when in this state? The last frame presented
+        to the user will not reflect the DOM state as it asynchronously switches to
+        the new version.
+
+        Note: The aim is to prevent unintended DOM updates from being presented to the
+        user after a cached snapshot for the elements has been captured. We wait for
+        one rendering opportunity after prepare to present DOM mutations made by the
+        author before prepare to be presented to the user. This is also the content
+        captured in snapshots.
+
+    Note: These steps will be added to the [=update the rendering=] in the HTML spec.
+        As such, the prose style is written to match other steps in that algorithm.
+</div>
+
 ## [=Perform pending transition operations=] ## {#perform-pending-transition-operations-algorithm}
 
 <div algorithm>
     To <dfn>perform pending transition operations</dfn> given a {{Document}} |document|,
         perform the following steps:
 
-    1. If |document|'s [=document/pending same document transition outgoing capture=] is not null, then:
-
-        1. [=Perform an outgoing capture=] with |document|'s [=document/pending same document transition outgoing capture=].
+    1. If |document|'s [=document/active DOM transition=] is not null, then:
 
-        1. Set |document|'s [=document/pending same document transition outgoing capture=] to null.
+        1. If |document|'s [=document/active DOM transition=]'s [=DOMTransition/phase=] is "`pending-capture`", then [=perform an outgoing capture=] with |document|'s [=document/active DOM transition=].
```

Comment:
I initially thought we need another phase between pending-capture and dom-update-callback-called. Since other tasks can run between the pending-capture finishing asynchronously and we don't want to run [=perform an outgoing capture=]twice.

But since document/transition suppressing rendering is set in perform an outgoing capture, update the rendering can only run once! We should add a note for that, it's quite subtle. :)

Reply:
Ehhhh, it's way more subtle than I intended. I think we should add another phase for this to make it clearer.

Reply:
(bless you for assuming I was being smart here rather than dumb 😄)